### PR TITLE
Python. Homework #1. Task #1. Мищенко Станислав Алексеевич

### PR DESCRIPTION
**Pull request by mi-sts**

### CODE REVIEW

File: src/hw1/vector.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+from src.util.type_checking import is_number
+
+
+class Vector:
+    def __init__(self, first, *args):
+        if not all(is_number(element) for element in args):
+            raise TypeError("A vector can't have a non-numeric element!")
+
+        self.elements = [first] + list(args)
+        self.length = 1 + len(args)
```
Comment (areyde) [inline src/hw1/vector.py]: This looks a bit awkward. Why do we need a specific "first"? Why not do just args, but add a check that it's not empty?

File: src/hw1/vector.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        elif isinstance(other, Vector):
+            if other.length != self.length:
+                raise ArithmeticError("Vectors in the multiplication have a different length!")
+
+            return sum([i * j for i, j in list(zip(self.elements, other.elements))])  # Scalar product.
+        else:
+            raise TypeError("A vector can only be multiplied by a number or a another vector!")
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
```
Comment (areyde) [inline src/hw1/vector.py:55]: What exactly does this do?

**Review by areyde — CHANGES_REQUESTED** [review]
Great work, nice tests 💪 A lot of extra stuff done.

File: src/hw1/vector.py (hunk 2 above)
Comment (mi-sts) [reply src/hw1/vector.py:55]: If an integer is to the left of * operator, then this function will be called, so I just redirect the call.

**Review by areyde — APPROVED** [review]
Nice! 💪

**Pull request merged**